tck/src/main/java/org/eclipse/microprofile/concurrency/tck/MPConfigTest.java

```diff
@@ -738,4 +743,121 @@ public void overrideMaxQueuedWith4ForManagedExecutorFieldWithConfigAndName()
             barrier.forceTermination();
         }
     }
+
+    /**
+     * CDI programmatic lookup of ManagedExecutor without any qualifiers must not find anything.
+     * This test is possible for ManagedExecutor because the test application does not provide any
+     * producers of ManagedExecutor without qualifiers.
+     */
+    @Test(dependsOnMethods = "beanInjected")
+    public void programmaticLookupManagedExecutorWithoutQualifiers() {
```

Comment:
we should have an equivalent test for ThreadContext as well.

Also, I believe it was agreed on that the container would *block* any attempts by the application to register an `@Default` producer. If I'm remembering that correctly, let me know and I"ll add a test scenario for it in the tracking work item.

Reply:
I couldn't add a test for ThreadContext to this test application because it already had a producer for ThreadContext with different qualifier, and that gets returned instead of indicating unsatisfied.  The select method must assume `@Any` when no qualifiers are present.

Reply:
And yes, I think you are remembering correctly about there being a decision to block/fail the application from registering a default producer for ManagedExecutor/ThreadContext, although I can't remember that actually being written down in the spec, so when we get to that, it may need a sentence added to JavaDoc or spec a well if it isn't already there.